internal/session/session.go

```diff
@@ -35,26 +35,22 @@ type PeerManager interface {
 	// interested in a peer's connection state
 	UnregisterSession(uint64)
 	// SendWants tells the PeerManager to send wants to the given peer
-	SendWants(ctx context.Context, peerId peer.ID, wantBlocks []cid.Cid, wantHaves []cid.Cid)
+	SendWants(sid uint64, peerId peer.ID, wantBlocks []cid.Cid, wantHaves []cid.Cid)
 	// BroadcastWantHaves sends want-haves to all connected peers (used for
 	// session discovery)
-	BroadcastWantHaves(context.Context, []cid.Cid)
+	BroadcastWantHaves(sid uint64, whs []cid.Cid)
 	// SendCancels tells the PeerManager to send cancels to all peers
-	SendCancels(context.Context, []cid.Cid)
+	SendCancels(sid uint64, cs []cid.Cid)
 }
 
 // SessionManager manages all the sessions
 type SessionManager interface {
 	// Remove a session (called when the session shuts down)
 	RemoveSession(sesid uint64)
-	// Cancel wants (called when a call to GetBlocks() is cancelled)
-	CancelSessionWants(sid uint64, wants []cid.Cid)
```

Comment:
The SessionManager no longer cancels wants - instead the Session tells the PeerManager that it's no longer interested in a want and the PeerManager keeps track of which sessions are interested in which wants